### PR TITLE
Added a simple circle class

### PR DESCRIPTION
**Pull request by arneboe**
Right now it only provides a simple intersection test but the implementation will grow over time, I guess.

### CODE REVIEW

File: src/Circle.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    /*Circles share centers. This results in division by zero,
+      infinite solutions or one circle being contained within the other. */
+    if(d == 0.0)
+        return result;
+    //Circles do not touch each other
+    else if(d > (r + r1))
+        return result;
+    //One circle is contained within the other
+    else if(d < (r - r1))
```
Comment (doudou) [inline src/Circle.cpp]: shouldn't that be `fabs(r - r1)` ?

File: src/Circle.cpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        return result;
+    }
+
+    //Get the perpendicular slope by multiplying by the negative reciprocal
+    //Then multiply by the h offset in relation to d to get the actual offsets
+    double mx = -(dy * h/d);
+    double my =  (dx * h/d);
+
+    //Add the offsets to point p2 to obtain the intersection points
+    result.emplace_back(px2 + mx, py2 + my);
```
Comment (doudou) [inline src/Circle.cpp]: This is C++11, on which no definite decision has been made for rock core packages (feel free to restart the discussion on rock-dev). Given that there would be no major improvement in this case over C++98's `push_back(Point(...))`, I would suggest to use the C++98 construct in the meantime.

File: src/Circle.cpp (hunk 1 above)
Comment (arneboe) [reply src/Circle.cpp]: Indeed :)

File: src/Circle.cpp (hunk 2 above)
Comment (arneboe) [reply src/Circle.cpp]: Ok :)

**Comment by arneboe** [comment]
Fixed

**Comment by chhtz** [comment]
Generally, this looks good to me. I'm not sure, if we really need to encapsulate the data members with getters and setters. In comparison, most types in `base/types` simply have public data members.

**Comment by jmachowinski** [comment]
The reason for this is that only classes without private members can be auto serialized by typelib / orogen.
If this class is intended, to ever go onto an port, we should remove the private:, I would leave the getters and setters though.

**Comment by arneboe** [comment]
The intention of this class is to act as a math library for all calculations involving circles.
It is not intended to go onto a port. However since a circle is very well defined i don't see any harm in removing the private. It's not like radius and center are ever gonna be removed. 
But we should be careful when adding additional attributes (e.g. lookup tables) in the future.

**Comment by chhtz** [comment]
I guess it will make sense to send it over ports, e.g., if you have a component that detects circles in camera images (I don't have a need for that right now). Also, I would avoid doing look-up tables in primitive types.

**Comment by arneboe** [comment]
Ok. I added a missing include and made the attributes public.
Should be fine to merge now.

**Pull request merged**